samples/boards/argonkey/src/main.c

```diff
@@ -0,0 +1,343 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+
+#include <board.h>
+#include <gpio.h>
+#include <i2c.h>
+#include <spi.h>
+#include <sensor.h>
+
+/* #define ARGONKEY_TEST_LOG 1 */
+
+#define LED0_GPIO_NAME		LED0_GPIO_CONTROLLER
+#define LED0_PIN		LED0_GPIO_PIN
+#define LED1_GPIO_NAME		LED1_GPIO_CONTROLLER
+#define LED1_PIN		LED1_GPIO_PIN
+
+#define WHOAMI_REG      0x0F
+#define WHOAMI_ALT_REG  0x4F
+
+static inline float out_ev(struct sensor_value *val)
+{
+	return (val->val1 + (float)val->val2 / 1000000);
+}
+
+static int lsm6dsl_trig_cnt;
+#ifdef CONFIG_LSM6DSL_TRIGGER
+static void lsm6dsl_trigger_handler(struct device *dev,
+				    struct sensor_trigger *trig)
+{
+#ifdef ARGONKEY_TEST_LOG
+	char out_str[64];
+#endif
+	struct sensor_value accel_x, accel_y, accel_z;
+	struct sensor_value gyro_x, gyro_y, gyro_z;
+
+#if defined(CONFIG_LSM6DSL_EXT0_LIS2MDL)
+	struct sensor_value magn_x, magn_y, magn_z;
+#endif
+#if defined(CONFIG_LSM6DSL_EXT0_LPS22HB)
+	struct sensor_value press, temp;
+#endif
+
+	lsm6dsl_trig_cnt++;
+
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_ACCEL_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_X, &accel_x);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Y, &accel_y);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Z, &accel_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "accel (%f %f %f) m/s2", out_ev(&accel_x),
+						out_ev(&accel_y),
+						out_ev(&accel_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+	/* lsm6dsl gyro */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_GYRO_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_X, &gyro_x);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Y, &gyro_y);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Z, &gyro_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "gyro (%f %f %f) dps", out_ev(&gyro_x),
+						out_ev(&gyro_y),
+						out_ev(&gyro_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+#if defined(CONFIG_LSM6DSL_EXT0_LIS2MDL)
+	/* lsm6dsl magn */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_MAGN_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_X, &magn_x);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_Y, &magn_y);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_Z, &magn_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "magn (%f %f %f) gauss", out_ev(&magn_x),
+						 out_ev(&magn_y),
+						 out_ev(&magn_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+#endif
+#if defined(CONFIG_LSM6DSL_EXT0_LPS22HB)
+	/* lsm6dsl press/temp */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_PRESS);
+	sensor_channel_get(dev, SENSOR_CHAN_PRESS, &press);
+
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_AMBIENT_TEMP);
+	sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &temp);
+
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "press (%f) kPa - temp (%f) deg", out_ev(&press),
+							   out_ev(&temp));
+	printk("%s\n", out_str);
+#endif
+
+#endif
+}
+#endif
+
+void main(void)
+{
+	int cnt = 0;
+	char out_str[64];
+
+#ifdef CONFIG_BOARD_96B_ARGONKEY
```

Comment:
This is a sample for the 96b_argonkey board why do you add `#ifdef CONFIG_BOARD_96B_ARGONKEY` block around the code which configures the board's leds?

Reply:
Because I built previously the same code for other platform too (e.g. nucleo f411). I forgot to delete it now that I decided to make it available for testing ArgonKey.
ACK